lib/spack/spack/binary_distribution.py

```diff
@@ -44,28 +44,47 @@
 import spack.relocate as relocate
 
 
-class NoOverwriteException(Exception):
-    pass
+class NoOverwriteException(spack.error.SpackError):
+    def __init__(self, filepath):
+        super(NoOverwriteException, self).__init__(
+            "%s exists, use -f to force overwrite." % filepath)
 
 
-class NoGpgException(Exception):
-    pass
+class NoGpgException(spack.error.SpackError):
+    def __init__(self):
+        super(NoGpgException, self).__init__(
+            "gpg2 is not available,"
+            " use -y to create unsigned build caches")
```

Comment:
Yes, the idea was to do a different PR to include it. Since it is possible to create and install a package in the buildcache without using gpg, IMO it should be able to do it also when using buildcache through `spack install`. Trying to do so, I hit the problem of no error message.

Reply:
GPG signatures really shouldn't be ignored when installing prebuilt packages. It certainly shouldn't be happening by default and I wouldn't mind seeing the flag to do so hidden from `--help` either.

Reply:
@mathstuf I agree, as long as those prebuilt packages are somehow distributed they should be signed. However, I find useful an option to ignore such gpg step when working locally just to reduce the time my stack gets to be installed or when working with a local mirror.